security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -749,6 +754,8 @@ func (sc *SecretManagerClient) UpdateConfigTrustBundle(trustBundle []byte) error
 	sc.configTrustBundleMutex.Unlock()
 	cacheLog.Debugf("update new trust bundle")
 	sc.OnSecretUpdate(security.RootCertReqResourceName)
+	sc.cache.SetWorkload(nil)
```

Comment:
workload's root cert might be different from XDS root cert, for example they might come from meshconfig, In this case, we should not trigger workload cert update if only xds root cert is rotated.

Reply:
when meshconfig changed, root cert rotated, still need to wait for 12h(in the worest) before workload cert rotate?

Reply:
@irisdingbj The trust bundle here is sent via PCDS, and actually it is from `s.workloadTrustBundle`. i am not sure how it is related to xds server root cert here.

For xds server root cert, the client use is mounted in file system. See `Agent.FindRootCAForXDS`

Reply:
When root cert for workload is coming from meshconfig instead of plugin root cert or selfsigned root cert, can this change cover that rotation as well?

Reply:
Yes, this is via `updatePluggedinRootCertAndGenKeyCert`